src/composables/accountSelector.ts

```diff
@@ -0,0 +1,200 @@
+  const ownAddresses = computed(() => {
+    if (protocolFilter.value) {
+      return accountsGroupedByProtocol.value[protocolFilter.value]?.map((account) => (
+        {
+          name: getName(account.address).value || getDefaultAccountLabel(account),
+          address: account.address,
+          isBookmarked: false,
+          protocol: protocolFilter.value ?? PROTOCOLS.aeternity,
+          type: account.type,
+        }
+      ));
+    }
+    return [];
+  });
+  const accountsFilteredByType = computed(
+    () => {
+      switch (accountSelectType.value) {
+        case ACCOUNT_SELECT_TYPE_FILTER.bookmarked:
+        case ACCOUNT_SELECT_TYPE_FILTER.addressBook:
+          return addressBookFiltered.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.recent:
+          return latestTransactions.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.owned:
+          return ownAddresses.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.all:
+          return [...(addressBookFiltered.value ?? []), ...(ownAddresses.value ?? [])];
+        default:
+          return [];
+      }
+    },
+  );
+  const accountsFiltered = computed(() => {
+    const entries: IAccountSelectorEntry[] = accountsFilteredByType.value ?? [];
+    const searchQueryLower = searchQuery.value.toLowerCase();
+
+    return entries.filter( // Filter by searchQuery
+      ({ name, address }) => [name, address].some(
+        (val) => val.toLowerCase().includes(searchQueryLower),
+      ),
+    ).filter( // Remove duplicates
+      (entry, index, self) => self.findIndex((e) => e.address === entry.address) === index,
+    );
+  });
+
+  function setAccountSelectType(type: AccountSelectTypeFilter, resetProtocolFilter = false) {
+    accountSelectType.value = type;
+    setShowBookmarked(type === ACCOUNT_SELECT_TYPE_FILTER.bookmarked, resetProtocolFilter);
+  }
+  function clearFilters(resetProtocolFilter = false) {
+    accountSelectType.value = ACCOUNT_SELECT_TYPE_FILTER.addressBook;
+    addressBookClearFilters(resetProtocolFilter);
+  }
+
+  watch(
+    protocolFilter,
+    () => {
+      if (protocolFilter.value) {
+        setAccountSelectType(ACCOUNT_SELECT_TYPE_FILTER.addressBook);
+      }
+    },
+  );
+  watch(
+    [accountsTransactionsLatest.value, activeAccount],
+    async () => {
+      latestTransactions.value = await Promise.all(
+        (accountsTransactionsLatest
+          .value[activeAccount.value.address] || []).filter(
+          (transaction: ITransaction) => {
+            const outerTx = transaction.tx!;
+            const innerTx = transaction.tx ? getInnerTransaction(transaction.tx) : null;
+            const { recipientId } = outerTx?.payerId ? outerTx : innerTx;
+            const { direction } = useTransactionData({
+              transaction: toRef(() => transaction),
+            });
+            return direction.value === TX_DIRECTION.sent && recipientId;
+          },
+        ).map(async (transaction: ITransaction): Promise<IAccountSelectorEntry> => {
```

Comment:
I strongly believe this action should not be asynchronous. See my comment regarding name resolving.